quinn-proto/src/connection/mod.rs

```diff
@@ -952,6 +1000,7 @@ where
                 self.spaces[space].pending_acks.subtract(&info.acks);
                 ack_eliciting_acked |= info.ack_eliciting;
                 self.on_packet_acked(now, info);
+                self.path.mtud.acked(packet);
```

Comment:
We probably should, but I figured the MVP could do without it. Would you want to shrink the default receive buffer size and resize it as soon the max datagram size increases?

Reply:
Sorry, I meant application datagram size. Strictly speaking I think MTU is a one-directional property, and in any case we set the receive buffer size based on the hard maximum we advertise support for. Leaving this for future work is fine; I only mention it since I expect it's only a few LoC to do so.